### PR TITLE
add express

### PR DESCRIPTION
**Pull request by lisperata**
image
image
image
image
image
image

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    res.status(500).send(err);
+  }
+});
+
+app.put("/put", async (req, res) => {
+  try {
+    const result: string = await putUser(
+      req.body.id,
+      req.body.name,
+      req.body.age
```
Comment (CatDidenko) [inline index.ts]: Use destructuring  http://es6-features.org/#ObjectMatchingShorthandNotation

File: models/model.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+          reject("Error 500. Internal Server Error");
+        }
+        resolve(content);
+      });
+    });
+  }
+
+  public addUser(): Promise<string> {
+    return new Promise((resolve, reject) => {
+      fs.readFile(pathToJsonFile, "utf-8", (err, content) => {
```
Comment (CatDidenko) [inline models/model.ts]: Use promise version on these functions https://nodejs.org/api/fs.html#fs_fspromises_readfile_path_options

File: index.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+app.use(bodyParser.urlencoded());
+
+app.get("/", async (req, res) => {
+  try {
+    const users = await getUsers();
+    res.type("json").send(users);
+  } catch (err) {
+    res.status(500).send(err);
+  }
```
Comment (CatDidenko) [inline index.ts]: Move logic with statuses to controller.

File: controller/controller.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  id: number,
+  name: string,
+  age: string
+): Promise<string> => {
+  const user = new User(id, name, age);
+  try {
+    return await user.addUser();
+  } catch (err) {
+    throw err;
+  }
```
Comment (CatDidenko) [inline controller/controller.ts]: Look at project structure https://github.com/hagopj13/node-express-boilerplate.
The controller should be responsible for returning the http response to the user.

File: index.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+});
+
+app.put("/put", async (req, res) => {
+  try {
+    const result: string = await putUser(
+      req.body.id,
+      req.body.name,
+      req.body.age
+    );
+    res.type("plain").status(201).send(result);
```
Comment (CatDidenko) [inline index.ts]: Is 201 status for put request?
https://stackoverflow.com/questions/2342579/http-status-code-for-update-and-delete

**Review by CatDidenko — CHANGES_REQUESTED** [review]

File: index.ts (hunk 5 above)
Comment (lisperata) [reply index.ts]: Changed

File: controller/controller.ts (hunk 4 above)
Comment (lisperata) [reply controller/controller.ts]: Changed

File: index.ts (hunk 3 above)
Comment (lisperata) [reply index.ts]: Logic with statuses has been moved to the controller

File: models/model.ts (hunk 2 above)
Comment (lisperata) [reply models/model.ts]: Used fs.promises

File: index.ts (hunk 1 above)
Comment (lisperata) [reply index.ts]: Destructured

**Pull request merged**